cli/flox-watchdog/src/process.rs

```diff
@@ -0,0 +1,139 @@
+use std::fs::read_to_string;
+
+use tracing::debug;
+
+/// https://man7.org/linux/man-pages/man5/proc_pid_stat.5.html
+/// Process has finished but the parent hasn't reaped it yet.
+const PROCFS_STATE_ZOMBIE: &str = "Z";
+/// Process is dead and will transition to a zombie or disappear.
+/// Technically we shouldn't see this, but just in case:
+/// https://unix.stackexchange.com/a/653370
+const PROCFS_STATE_DEAD: &str = "X";
+const PROCFS_STATE_DEAD_COMPAT: &str = "x";
+
+/// Process watcher for Linux based on `procfs`.
+#[derive(Debug, Clone, PartialEq)]
+pub(crate) struct ProcfsWatcher {
+    pid: i32,
+}
+
+/// Allow compilation when not on Linux to make IDE/LSP development easier. It
+/// will always return negative results when not used on other platforms.
+#[cfg_attr(not(target_os = "linux"), allow(dead_code))]
+impl ProcfsWatcher {
+    /// The PID will *not* be checked on instantiation so that we don't need to
+    /// do additional cleanup checks before `wait_for_termination`.
+    pub(crate) fn new(pid: i32) -> Self {
+        Self { pid }
+    }
+
+    /// Reads the state of a process. Any failure is used as an indication that
+    /// the process is no longer running.
+    fn state(&self) -> Option<String> {
+        let path = format!("/proc/{}/stat", self.pid);
+        let stat = match read_to_string(path) {
+            Ok(stat) => stat,
+            Err(err) => {
+                debug!(%err, self.pid, "failed to read stat, treating as dead");
+                return None;
+            },
+        };
+        let state = match stat.split_whitespace().nth(2) {
+            Some(state) => state.to_string(),
+            None => {
+                debug!(self.pid, "failed to parse stat, treating as dead");
```

Comment:
From sync with Yannik: this is unexpected, whereas not being able to read the file above is expected when the PID has gone:
```suggestion
                warn!(self.pid, "failed to parse stat, treating as dead");
```

Reply:
Done